RoomMate.Domain/Services/Implements/RegisterService.cs

```diff
@@ -3,58 +3,77 @@
 using System;
+using System.Collections.Generic;
 
 namespace RoomMate.Domain.Services.Implements
 {
 
     public class RegisterService : IRegisterService
     {
         private readonly IRepository<User> _userRepository;
+        private readonly IAddressService  _addressService;
 
 
-        public RegisterService(IRepository<User> userRepository)
+        public RegisterService(IRepository<User> userRepository,
+             IAddressService addressService)
         {
             this._userRepository = userRepository;
+            //what with dependency? added to service addresDepencendy
+            this._addressService = addressService;
         }
 
 
         public bool RegisterUser(RegisterDto registerDto)
         {
+            bool isRegistered = false;
+            int idOfJustCreatedUser = 0;
+
             //check unique values of login and email
             if (IsUserTaken(registerDto.Login, registerDto.Email))
             {
-                return false;
+                return isRegistered;
             }
-            //tu mielonka dto przekazuje z register dto i potem wale tym do repo?
-            //w repo jeszcze robie czysty obiekt do inserta do tabeli
             //crypting password
             var crypto = new SimpleCrypto.PBKDF2();
             registerDto.Password = crypto.Compute(registerDto.Password); //crypted pass
             registerDto.PasswordSalt = crypto.Salt; //salt
-            
             //add missing parts of user to registerDto and pass it to the convertToTarget
             registerDto.Active = true;
             registerDto.Code = ""; //usefull on reset password only
             registerDto.CreatedBy = null;
             registerDto.CreatedDate = DateTime.Now;
             registerDto.ModificatedDate = DateTime.Now;
             registerDto.ModificatedBy = null;
-
             //add user to db
             try
             {
                 var user = this.ConvertToTarget(registerDto);
                 this._userRepository.InsertOrUpdate(user);
-                return true;
+                isRegistered =  true;
+                idOfJustCreatedUser = user.Id;
 
             }
             catch (Exception ex)
             {
-                return false;
+                isRegistered =  false;
             }
-        }
+           
+            //tu o metoda do inputow w tabeli adres i polaczyc to z userem
+            // onyly if user seleceted valid addres
+            //if user is registered correctly then insert the address otherwise thers no point
+            if (ValidationDto.IsValidId(registerDto.AddressDto.Id) && isRegistered == true)
+            {
+                isRegistered = _addressService.AddAddressUserSelected(registerDto.AddressDto, idOfJustCreatedUser);
+            }
+            //if user selected only city
+            else if (!ValidationDto.IsValidId(registerDto.AddressDto.Id)
+                && ValidationDto.IsValidId(registerDto.AddressDto.CityId))
+            {
 
```

Comment:
A to co?

Reply:
Registration including two use cases, one if user selected city second: address and city